### ISSUE TITLE
Kinetic release

### ISSUE THREAD

**Issue by nlyubova** [ros-naoqi/nao_virtual#19]
could we release it for Jade and Kinetic?

**Comment by mikaelarguedas** [comment]
Now that moveit has been released in Kinetic we could start porting these packages indeed.
Only blocker so far: both jade an kinetic are using Gazebo5+ and `roboticsgroup_gazebo_plugins` are using gazebo2 API so won't compile against gazebo5 and more recent.

Right now Nao is not stable in gazebo on kinetic (once removed every dependency on `roboticsgroup_gazebo_plugins`). I'll release these once this is fixed

**Comment by blackvitriol** [comment]
Thank you for this information. Awaiting kinetic release, patiently.
*drowns in pending research anxiety*

**Comment by mikaelarguedas** [comment]
I don't know when I'll have time to work on this. Maybe @nlyubova will be able to look into it in the  foreseeable future?

**Comment by gortium** [comment]
+1 would like a kinetic release too ^^

**Comment by blackvitriol** [comment]
Miss Lyubova, can you please state which kind of dependencies are not supported by Kinetic. I have to find a way around it.

**Comment by mikaelarguedas** [comment]
the dependency that was not supported is `roboticsgroup_gazebo_plugins`. It should now be building against gazebo7 thanks to the updated API from [this PR](https://github.com/roboticsgroup/roboticsgroup_gazebo_plugins/pull/7). From what I remember there were some weird physics behaviour (robot floating around instead of falling etc). I didn't have time to look at it since then, so if you find what is causing this weird behavior we would be very happy to review and merge the appropriate fix.

**Comment by blackvitriol** [comment]
What about apt-get install ros-**kinetic**-nao-gazebo-plugin ?

**Comment by mikaelarguedas** [comment]
> What about apt-get install ros-kinetic-nao-gazebo-plugin ?

Yes that is the command you will be able to run once the package is released. But the simulation needs to be working properly before the release is made.

Did you try building from source ? Can you reproduce the strange behaviour of the robot floating around?

**Comment by blackvitriol** [comment]
> Did you try building from source ? Can you reproduce the strange behaviour of the robot floating around?

Sorry I am not there yet, I plan on doing this now and to debug all problems by myself. Any help/ input is appreciated.

**Comment by gortium** [comment]
The floating robot is due to the position joint controller/hardware interfaces as spoken there: 
https://github.com/ros-simulation/gazebo_ros_pkgs/issues/479

We need to use positionPID or velocity interface. So change in the nao_description and nao_control are needed. To be compatible with real NAOs there is possibly some changes in naoqi too as it does not support velocity control of the joints! That could help on that:
https://community.ald.softbankrobotics.com/en/forum/visual-servoing-nao-using-ros-12212

**Comment by amstrudy** [comment]
Is there any update on getting a kinetic release?

**Comment by ahmedius2** [comment]
I compiled Gazebo 7.0.0 from source by changing the code using this [pull request](https://bitbucket.org/vincentrou/gazebo/commits/8d87bb730418e06e5eecd743cf54a812744d500a?at=default) and the floating nao problem is gone now. I don't know if it is a good solution but it solves the problem and you can run the simulation in kinetic.

**Comment by mikaelarguedas** [comment]
thanks @ahmedius2 for the info!
It looks like this commit never got merged and released into gazebo7. I see another Pull Request allegedly addressing the same issue [here](https://bitbucket.org/osrf/gazebo/pull-requests/2814). This one has been merged into Gazebo9 but hasn't been backported to gazebo7 (likely because it breaks API). As ROS Kinetic and Lunar are using gazebo7 from Ubuntu, we will need a solution that works with that version. 

Apparently vincetrou found a workaround in setting PID values explicitly (https://github.com/ros-simulation/gazebo_ros_pkgs/issues/479#issuecomment-319610133). Maybe this would be a good alternative to try here?

**Comment by hayday100** [comment]
@ahmedius2 Could you provide details on how to get Virtual NAO to work in Kinetic?  Thanks in advance!

**Comment by leyongloh** [comment]
my solution: 
git clone -b gazebo7 https://github.com/pal-robotics/pal_gazebo_plugins.git 
and follow the step from readme to run the simulation
You might encounter some error. For example,
-missing dependencies: ddynamic-reconfigure
-apt-get install ros-kinetic-ddynamic-reconfigure
